centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -483,9 +483,102 @@ object Operations extends StrictLogging {
     } yield selected
   }
 
+  private def removeMetadataSourceFromJsObject(jsObject: JsObject) = {
+    val metadataSourceKeyName = "metadataSource"
+    JsObject(jsObject.fields - metadataSourceKeyName)
+  }
+
+  def compareMetadataBeforeAndAfterArchival(submittedWorkflow: SubmittedWorkflow,
+                                            workflow: Workflow,
+                                            expandSubworkflows: Boolean): Test[Unit] = new Test[Unit] {
+
+    override def run: IO[Unit] = {
+      for {
+        metadataBeforeArchival <- CentaurCromwellClient.metadata(submittedWorkflow, expandSubworkflows = expandSubworkflows, archived = Option(false))
+        metadataAfterArchival <- CentaurCromwellClient.metadata(submittedWorkflow, expandSubworkflows = expandSubworkflows, archived = Option(true))
+        metadataObjectBeforeArchival <- IO.fromTry(Try(removeMetadataSourceFromJsObject(metadataBeforeArchival.value.parseJson.asJsObject)))
+        metadataObjectAfterArchival <- IO.fromTry(Try(removeMetadataSourceFromJsObject(metadataAfterArchival.value.parseJson.asJsObject)))
+        _ <- validateMetadataJson(metadataObjectBeforeArchival, metadataObjectAfterArchival, submittedWorkflow, workflow)
+      } yield ()
+    }
+  }
+
+  def compareLabelsBeforeAndAfterArchival(labelsLikelyBeforeArchival: IO[WorkflowLabels],
+                                          labelsAfterArchival: IO[WorkflowLabels],
+                                          submittedWorkflow: SubmittedWorkflow,
+                                          workflow: Workflow): Test[Unit] = new Test[Unit] {
+
+    override def run: IO[Unit] = {
+      import centaur.test.metadata.WorkflowFlatLabels._
+      for {
+        labelsLikelyBeforeArchivalFlat <- labelsLikelyBeforeArchival map { _.asFlat.stringifyValues }
+        labelsAfterArchivalFlat <- labelsAfterArchival map { _.asFlat.stringifyValues }
+        _ <- compareFlatMetadata(
+          expected = labelsLikelyBeforeArchivalFlat,
+          actual = labelsAfterArchivalFlat,
+          // here we use generic type instead of `LabelsFlatMetadataType` because we need to do  full comparison, i.e.
+          // expected.diff(actual) and actual.diff(expected), unlike the regular `validateMetadata` test where we only
+          // check that actual labels contain all labels specified in the *.test file
+          GenericFlatMetadataType("labels"),
```

Comment:
Is there a reason you're comparing flattening and only comparing the flat metadata here, instead of comparing them as  `JsObject`s?